cursor.go

```diff
@@ -375,67 +399,173 @@ func (c *cursor) CursorFilterSortDirection(naturalSortDirection int) int {
 	return naturalSortDirection * -1
 }
 
-// SetTimeCursorFilter looks for a field named fieldName in the cursor and assumes it is a time.Time
-func (c *cursor) SetTimeCursorFilter(findFilter bson.M, fieldName string, naturalSortDirection int) (err error) {
+func (c *cursor) AddCursorFilters(findFilter bson.M, naturalSortDirection int, filterFields ...CursorFilterField) (err error) {
 	cursorValues := c.cursorFilter()
 	if len(cursorValues) == 0 {
 		// if no cursor specified do nothing and that's ok
 		return
 	}
 	// Note: If we have a cursor it should be a valid one so error after here
 
-	if fieldValue, ok := cursorValues[fieldName]; ok {
-		timeValue := time.Now()
-		err = timeValue.UnmarshalText([]byte(fieldValue))
+	var filters []bson.M
+	for i, filterField := range filterFields {
+		var newFilterFieldValue interface{}
+		newFilterFieldValue, err = c.UnmarshalFieldValue(filterField)
+		if err != nil {
+			return
+		}
+
+		newFilter := bson.M{
+			filterField.FieldName: bson.D{{c.cursorFilterOperator(naturalSortDirection), newFilterFieldValue}},
```

Comment:
just to be consistent
```suggestion
			filterField.FieldName: bson.M{c.cursorFilterOperator(naturalSortDirection): newFilterFieldValue},
```
so combined with change above that would result in:
```
expectedFilter := primitive.M{
		"$or": primitive.A{
			primitive.M{
				"created": primitive.M{
					"$gt": createdTime,
				},
			},
			primitive.M{
				"created": createdTime,
				"oid": primitive.M{
					"$gt": mongoOid,
				},
			},
		},
	}
```